### PR TITLE
SNOW-209092 Skip Session Close

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
### Description
Added a control flow to keep client session alive if parameter enabled.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by sfc-gh-anezvigin** [comment]
My concern with using `isClientSessionKeepAliveEnabled()` is that if you turn on keep-alive, there's no way to close the session even if you wanted to.

**Comment by sfc-gh-anezvigin** [comment]
Also this is a pretty big behavior change for existing users of the driver that are turning off keep alive. Their sessions will stay open on the GS side.

**Comment by sfc-gh-jbahk** [comment]
I was thinking that users already enable the `client_session_keep_alive` to have the session persist and turn it off, followed by Close() to properly close the session. What do you think?

**Comment by sfc-gh-cshi** [comment]
Seems we don't have end to end test for this session persistence change for go driver. Should we have one?

**Comment by sfc-gh-anezvigin** [comment]
> I was thinking that users already enable the client_session_keep_alive to have the session persist and turn it off, followed by Close() to properly close the session. What do you think?

Problem is that Close() does more than just optionally close the snowflake session. Go's SQL interface uses it to clean up resources (eg: open channels, etc) that it uses to manage the query lifecycle. The user must call Close() when using the driver in long running processes.

So the following scenario would create a resource leak:

- Long running process (eg: API server)
- Sets client_session_keep_alive to false
- Does _not_ call Close() on the sql.DB object

**Review by sfc-gh-anezvigin — APPROVED** [review]
👍

File: connection.go [hunk 1]
```diff
@@ -203,9 +203,11 @@ func (sc *snowflakeConn) Close() (err error) {
 	logger.WithContext(sc.ctx).Infoln("Close")
 	sc.stopHeartBeat()
 
-	err = sc.rest.FuncCloseSession(sc.ctx, sc.rest, sc.rest.RequestTimeout)
-	if err != nil {
-		logger.Error(err)
+	if !sc.cfg.KeepSessionAlive {
+		err = sc.rest.FuncCloseSession(context.TODO(), sc.rest, sc.rest.RequestTimeout)
```
Comment (sfc-gh-dwu) [inline connection.go:207]: `context.Todo()` or should this be `sc.ctx`

**Review by sfc-gh-dwu — APPROVED** [review]
lgtm with 1 ctx change

**Pull request merged**